preparedstatement/src/main/java/software/amazon/athena/preparedstatement/CreateHandler.java

```diff
@@ -0,0 +1,79 @@
+package software.amazon.athena.preparedstatement;
+
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.services.athena.AthenaClient;
+import software.amazon.awssdk.services.athena.model.CreatePreparedStatementResponse;
+import software.amazon.awssdk.services.athena.model.InvalidRequestException;
+import software.amazon.awssdk.services.athena.model.ResourceNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class CreateHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<AthenaClient> proxyClient,
+        final Logger logger) {
+
+        this.logger = logger;
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+
+            .then(progress ->
+                proxy.initiate("AWS-Athena-PreparedStatement::Create::PreExistanceCheck",
+                    proxyClient, progress.getResourceModel(), progress.getCallbackContext())
+
+                    .translateToServiceRequest(Translator::translateToReadRequest)
+
+                    .makeServiceCall((awsRequest, client) -> {
+                        client.injectCredentialsAndInvokeV2(
+                            awsRequest, client.client()::getPreparedStatement);
+                        logger.log(String.format("%s has been read successfully.", ResourceModel.TYPE_NAME));
+                        throw new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, awsRequest.statementName());
+                    })
+
+                    .handleError((awsRequest, exception, client, model, context) -> {
+                        if (exception instanceof ResourceNotFoundException) { // no preparedstatement
+                            return ProgressEvent.progress(model, context);
+                        } else if (exception instanceof InvalidRequestException &&
+                            exception.getMessage().contains("is not found")) { // no workgroup
```

Comment:
Change the comment here. Also, are we sure we want to depend on the error message here? What is the expected behavior from GetPreparedStatement when Statement does not exist?

Reply:
when statement does not exist, the current behavior from GetPreparedStatement is ResourceNotFoundException, but when the workgroup does not exist, the current behavior is InvalidRequestException. We need a way to tell the no workgroup case from other invalidRequestException cases, since we want to handle it differently.